src/util/check.h

```diff
@@ -95,4 +95,12 @@ T&& inline_assertion_check(LIFETIMEBOUND T&& val, [[maybe_unused]] const char* f
         format_internal_error("Unreachable code reached (non-fatal)", \
                               __FILE__, __LINE__, __func__, PACKAGE_BUGREPORT))
 
+[[noreturn]] void inline_unreachable(const char* file, int line, const char* func);
```

Comment:
nit: I don't think this is inline. Maybe call it `unreachable_fail` for symmetry?

Reply:
Thanks! [Updated](https://github.com/bitcoin/bitcoin/pull/26504#issuecomment-1317292167).